test/tests/carousel_carousel-2-tablist.js

```diff
@@ -0,0 +1,450 @@
+      await waitAndCheckAriaSelected(t, index - 1),
+      'left arrow on tab "' + index + '" should set aria-selected="true" on previous tab.'
+    );
+    t.true(
+      await tabpanels[index - 1].isDisplayed(),
+      'left arrow on tab "' + index + '" should display the next previous panel.'
+    );
+  }
+});
+
+ariaTest('HOME key moves focus and selects tab', exampleFile, 'key-home', async (t) => {
+  t.plan(3 * ex.tabCount);
+
+  const tabs = await t.context.session.findElements(By.css(ex.tabSelector));
+  const tabpanels = await t.context.session.findElements(By.css(ex.tabpanelSelector));
+  for (let index = 0; index < tabs.length; index++) {
+
+    // Put focus on the tab
+    await openTabAtIndex(t, index);
+
+    // Send the home key to the tab
+    await tabs[index].sendKeys(Key.HOME);
+
+    // Check the focus is correct
+    t.true(
+      await waitAndCheckFocus(t, ex.tabSelector, 0),
+      'home key on tab "' + index + '" should put focus on the first tab.'
+    );
+    t.true(
+      await waitAndCheckAriaSelected(t, 0),
+      'home key on tab "' + index + '" should set aria-selected="true" on the first tab.'
+    );
+    t.true(
+      await tabpanels[0].isDisplayed(),
+      'home key on tab "' + index + '" should display the first tab.'
+    );
+  }
+});
+
+ariaTest('END key moves focus and selects tab', exampleFile, 'key-end', async (t) => {
+  t.plan(3 * ex.tabCount);
+
+  const tabs = await t.context.session.findElements(By.css(ex.tabSelector));
+  const tabpanels = await t.context.session.findElements(By.css(ex.tabpanelSelector));
+  for (let index = 0; index < tabs.length; index++) {
+
+    // Put focus on the tab
+    await openTabAtIndex(t, index);
+
+    // Send the end key to the tab
+    await tabs[index].sendKeys(Key.END);
+
+    // Check the focus is correct
+    t.true(
+      await waitAndCheckFocus(t, ex.tabSelector, tabs.length - 1),
+      'home key on tab "' + index + '" should put focus on the last tab.'
+    );
+    t.true(
+      await waitAndCheckAriaSelected(t, tabs.length - 1),
+      'home key on tab "' + index + '" should set aria-selected="true" on the last tab.'
+    );
+    t.true(
+      await tabpanels[tabs.length - 1].isDisplayed(),
+      'home key on tab "' + index + '" should display the last tab.'
+    );
+  }
+});
+
+// Keyboard interaction
+
+ariaTest('TAB moves key through buttons', exampleFile, 'rotation-key-tab', async (t) => {
+  t.plan(1);
+
+  await assertTabOrder(t, ex.allFocusableItems);
+});
+
+ariaTest('ENTER pause and start carousel motion', exampleFile, 'rotation-enter-or-space-toggle', async (t) => {
+  t.plan(2);
+
+  let activeElement = await t.context.session.findElement(By.css(ex.activeCarouselItem)).getAttribute('aria-label');
+
+  await t.context.session.findElement(By.css(ex.pausePlayButtonSelector)).sendKeys(Key.ENTER);
+  // Move focus from widget
+  await t.context.session.findElement(By.css(ex.pausePlayButtonSelector)).sendKeys(Key.chord(Key.SHIFT, Key.TAB));
```

Comment:
Is there a reason to sent shift+tab to the play button? Focus on the button doesn't currently prevent the carousel from playing -- if that is the intended behavior, it could help to add a test without shift+tab to verify it